common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -933,17 +935,36 @@ def _load_items(self, course_key, items, depth=0, using_descriptor_system=None):
                 course_key,
                 item,
                 data_cache,
-                apply_cached_metadata=(item['location']['category'] != 'course' or depth != 0),
-                using_descriptor_system=using_descriptor_system
+                using_descriptor_system=using_descriptor_system,
+                apply_cached_metadata=self._should_apply_cached_metadata(item, depth)
             )
             for item in items
         ]
 
+    def _should_apply_cached_metadata(self, item, depth):
+        """
+        Returns a boolean whether a particular query should trigger an application
+        of inherited metadata onto the item
+        """
+        category = item['location']['category']
+        apply_cached_metadata = category not in _DETACHED_CATEGORIES and \
+            not (category == 'course' and depth == 0)
+        return apply_cached_metadata
+
     @autoretry_read()
     def get_courses(self, **kwargs):
         '''
-        Returns a list of course descriptors.
+        Returns a list of course descriptors. This accepts an optional parameter of 'org' which
+        will apply an efficient filter to only get courses with the specified ORG
```

Comment:
Optional: It might be nice to have `org` as an explicit keyword arg in the function signature, since you've made it something supported by all modulestores.